### PR TITLE
Build fixes

### PR DESCRIPTION
**Pull request by hardening**

### CODE REVIEW

File: client/X11/xf_input.c [hunk 1]
```diff
@@ -875,17 +877,11 @@ static int xf_input_handle_event_remote(xfContext* xfc, const XEvent* event)
 				/* fallthrough */
 				WINPR_FALLTHROUGH
 			case XI_TouchUpdate:
-				/* fallthrough */
-				WINPR_FALLTHROUGH
```
Comment (akallabeth) [inline client/X11/xf_input.c:879]: what not working here?

**Review by akallabeth — CHANGES_REQUESTED** [review]
if a macro `WINPR_FALLTHROUGH` does not work for you then fix the error there and do not remove it in code.
should already define to empty if the compiler does not support the `c11` features

File: client/X11/xf_input.c (hunk 1 above)
Comment (hardening) [reply client/X11/xf_input.c:879]: Well it seems like for gcc a fallthrough is when you have
```C
case item:
    ... code...
   <no break>
case item 2:
```
The fallthrough is only at the point where it's missing a break, other cases are just regular `switch/case` matches (ie when there's no code between `case`)

Comment (mfleisz) [reply client/X11/xf_input.c:879]: Looks like this is actually the standard way of using fallthrough: https://en.cppreference.com/w/c/language/attributes/fallthrough

**Pull request merged**